src/Dapr.Actors.Analyzers/ActorAnalyzer.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Dapr.Actors.Analyzers;
+
+/// <summary>
+/// Analyzes actor registration in Dapr applications.
+/// </summary>
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class ActorAnalyzer : DiagnosticAnalyzer
+{
+    private static readonly DiagnosticDescriptor DiagnosticDescriptorActorRegistration = new(
+        "DAPR0001",
+        "Actor class not registered",
+        "The actor class '{0}' is not registered",
```

Comment:
We'll likely need to revisit this in the future as there's consideration for facilitating cross-service actor invocation, so it might very well be the case that an actor is referenced that doesn't exist in the same service.